cli/src/cmd/forge/test.rs

```diff
@@ -507,16 +507,16 @@ fn test(
     gas_reporting: bool,
 ) -> eyre::Result<TestOutcome> {
     if runner.count_filtered_tests(&filter) == 0 {
-        let mut err = String::from("No matching tests!");
+        let mut msg = String::from("No matching tests!");
         // Try to suggest a test when there's no match
         if let Some(ref test_pattern) = filter.test_pattern {
             let test_name = test_pattern.as_str();
             let candidates = runner.get_tests(&filter);
             if let Some(suggestion) = suggestions::did_you_mean(test_name, &candidates).pop() {
-                err = format!("{}\n\nDid you mean \"{}\"?", err, suggestion);
+                msg = format!("{}\n\nDid you mean \"{}\"?", msg, suggestion);
             }
         }
-        eyre::bail!(err)
+        println!("{}", msg);
```

Comment:
there's no easy way to get the match input? so that we could `no tests match filter: ...`

Reply:
if you're refering to (name/contract/path) patterns that the user provided, yes, it's possible to get those with `pattern.as_str()`

as for displaying, i think only showing patterns that users provided will do, e.g. for `forge test --match-test "test*" --no-match-contract ".*TestAvax"`, the message will be something like
```
No tests match the provided pattern:
match-test: test*
no-match-contract: .*TestAvax
```

did i understand you correctly?